command/profile/use_profile.go

```diff
@@ -0,0 +1,350 @@
+				if err != nil {
+					fmt.Printf("Bad github login for profile %s", name)
+				}
+			}
+			v.SetToken(profileCache.Vault.Auth.Token)
+			fmt.Printf("export VAULT_TOKEN=%s\n", profileCache.Vault.Auth.Token)
+			// TODO: More Auth methods to be added here
+		}
+	}
+
+	if profile.Consul.Server != "" {
+		fmt.Printf("export CONSUL_HTTP_ADDR=%s\n", profile.Consul.Server)
+	}
+
+	if profile.Consul.Auth.Token != "" {
+		fmt.Printf("export CONSUL_HTTP_TOKEN=%s\n", profile.Consul.Auth.Token)
+	}
+
+	if profile.Consul.Auth.Method == "vault" {
+		if profileCache.Consul.Auth.Token != "" {
+			if profileCache.Consul.Auth.ExpireTime != "" {
+				et, err := time.Parse(time.RFC3339Nano, profileCache.Consul.Auth.ExpireTime)
+				if err != nil { // if expire time  can't be parsed as time assume cache is bad and relogin
+					profileCache.Consul, err = vaultGetConsulCreds(profile.Consul, v)
+					if err != nil {
+						fmt.Errorf("Error reading Consul creds for %s profile ", name)
+					}
+				} else if et.Before(time.Now()) { // relogin if token is expired
+					profileCache.Consul, err = vaultGetConsulCreds(profile.Consul, v)
+					if err != nil {
+						fmt.Errorf("Error reading Consul creds for %s profile ", name)
+					}
+				}
+			}
+		} else { // token  is absent cache - login
+			profileCache.Nomad, err = vaultGetNomadCreds(profile.Nomad, v)
+			if err != nil {
+				fmt.Errorf("Error reading Consul creds for %s profile ", name)
+			}
+		}
+		fmt.Printf("export CONSUL_HTTP_TOKEN=%s\n", profileCache.Consul.Auth.Token)
+	}
+
+	if profile.Nomad.Server != "" {
+		fmt.Printf("export NOMAD_ADDR=%s\n", profile.Nomad.Server)
+	}
+
+	if profile.Nomad.Auth.Token != "" {
+		fmt.Printf("export NOMAD_TOKEN=%s\n", profile.Nomad.Auth.Token)
+	}
+
+	if profile.Nomad.Auth.Method == "vault" {
+		if profileCache.Nomad.Auth.Token != "" {
+			if profileCache.Nomad.Auth.ExpireTime != "" {
+				et, err := time.Parse(time.RFC3339Nano, profileCache.Nomad.Auth.ExpireTime)
+				if err != nil { // if expire time  can't be parsed as time assume cache is bad and relogin
+					profileCache.Nomad, err = vaultGetNomadCreds(profile.Nomad, v)
+					if err != nil {
+						fmt.Errorf("Bad reading of nomad creds for %s profile ", name)
+					}
+				} else if et.Before(time.Now()) { // relogin if token is expired
+					profileCache.Nomad, err = vaultGetNomadCreds(profile.Nomad, v)
+					if err != nil {
+						fmt.Errorf("Bad reading of nomad creds for %s profile ", name)
+					}
+				}
+			}
+		} else { // token  is absent cache - login
+			profileCache.Nomad, err = vaultGetNomadCreds(profile.Nomad, v)
+			if err != nil {
+				fmt.Errorf("Bad reading of nomad creds for %s profile ", name)
+			}
+		}
+		fmt.Printf("export NOMAD_TOKEN=%s\n", profileCache.Nomad.Auth.Token)
+	}
+
+	profilesCache[name] = profileCache
+
+	// Create a file for cache update
+	cacheTempFile, err := ioutil.TempFile("", "hashi_helper_cache")
+	if err != nil {
+		panic(err)
```

Comment:
why `panic` instead of returning the error?

Reply:
fixed

Reply:
Not really, it still says `panic`